db-engine/src/main/java/com/splicemachine/db/impl/sql/GenericStorablePreparedStatement.java

```diff
@@ -147,12 +148,15 @@ public ExecPreparedStatement getClone() throws StandardException {
     //
     /////////////////////////////////////////////////////////////
 
+    @SuppressFBWarnings("DMI_NONSERIALIZABLE_OBJECT_WRITTEN") // todo in DB-10583
     @Override
     public void writeExternal(ObjectOutput out) throws IOException {
+
+        // DANGER: do NOT change this serialization unless you have an upgrade script, see DB-10566
         out.writeObject(getCursorInfo());
         out.writeBoolean(needsSavepoint());
         out.writeBoolean(isAtomic);
-        out.writeObject(executionConstants);
+        out.writeObject(executionConstants); // DB-10583
```

Comment:
Yes, this is strange, here is an example of `ConstantAction` implementation that doesn't implement the `Externalizable` interface

![image](https://user-images.githubusercontent.com/52752081/97283537-40db0080-1840-11eb-9d91-a08ce34cca1f.png)